[CI] CustomAuthorizationEngine.java:50: error: cannot find symbol authentication.getUser().isRunAs()

### CI Link

https://gradle-enterprise.elastic.co/s/ozb6lycxc43es/console-log?task=:security-authorization-engine:compileJava

### Repro line

cd plugin/examples; ./gradlew build --include-build ../..

### Does it reproduce?

Yes

### Applicable branches

master

### Failure history

_No response_

### Failure excerpt

After merging https://github.com/elastic/elasticsearch/pull/86246 example builds started failing with compilation errors 

```
> Task :security-authorization-engine:compileJava FAILED
/Users/rene/dev/elasticsearch-master/plugins/examples/security-authorization-engine/src/main/java/org/elasticsearch/example/CustomAuthorizationEngine.java:50: error: cannot find symbol
        if (authentication.getUser().isRunAs()) {
                                    ^
  symbol:   method isRunAs()
  location: class User
/Users/rene/dev/elasticsearch-master/plugins/examples/security-authorization-engine/src/main/java/org/elasticsearch/example/CustomAuthorizationEngine.java:52: error: cannot find symbol
                new CustomAuthorizationInfo(authentication.getUser().authenticatedUser().roles(), null);
                                                                    ^
  symbol:   method authenticatedUser()
  location: class User
/Users/rene/dev/elasticsearch-master/plugins/examples/security-authorization-engine/src/main/java/org/elasticsearch/example/CustomAuthorizationEngine.java:61: error: cannot find symbol
        if (isSuperuser(requestInfo.getAuthentication().getUser().authenticatedUser())) {
                                                                 ^
  symbol:   method authenticatedUser()
  location: class User
Note: /Users/rene/dev/elasticsearch-master/plugins/examples/security-authorization-engine/src/main/java/org/elasticsearch/example/CustomAuthorizationEngine.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
3 errors

```

--- a/plugins/examples/security-authorization-engine/src/main/java/org/elasticsearch/example/CustomAuthorizationEngine.java
+++ b/plugins/examples/security-authorization-engine/src/main/java/org/elasticsearch/example/CustomAuthorizationEngine.java
@@ -47,9 +47,9 @@ public class CustomAuthorizationEngine implements AuthorizationEngine {
     @Override
     public void resolveAuthorizationInfo(RequestInfo requestInfo, ActionListener<AuthorizationInfo> listener) {
         final Authentication authentication = requestInfo.getAuthentication();
-        if (authentication.getUser().isRunAs()) {
+        if (authentication.isRunAs()) {
             final CustomAuthorizationInfo authenticatedUserAuthzInfo =
-                new CustomAuthorizationInfo(authentication.getUser().authenticatedUser().roles(), null);
+                new CustomAuthorizationInfo(authentication.getAuthenticatingSubject().getUser().roles(), null);
             listener.onResponse(new CustomAuthorizationInfo(authentication.getUser().roles(), authenticatedUserAuthzInfo));
         } else {
             listener.onResponse(new CustomAuthorizationInfo(authentication.getUser().roles(), null));
@@ -58,7 +58,7 @@ public class CustomAuthorizationEngine implements AuthorizationEngine {
 
     @Override
     public void authorizeRunAs(RequestInfo requestInfo, AuthorizationInfo authorizationInfo, ActionListener<AuthorizationResult> listener) {
-        if (isSuperuser(requestInfo.getAuthentication().getUser().authenticatedUser())) {
+        if (isSuperuser(requestInfo.getAuthentication().getAuthenticatingSubject().getUser())) {
             listener.onResponse(AuthorizationResult.granted());
         } else {
             listener.onResponse(AuthorizationResult.deny());

